### ISSUE TITLE
WebGL samples no longer work on Windows

### ISSUE THREAD

**Issue by cx20** [google/filament#3308]
**Describe the bug**
I tried to view a sample of Filament tutorial_triangle in Chrome 87.
However, an error appeared in the console and no triangles were displayed on the screen.

**To Reproduce**
1. Launch Chrome 87.
2. Go to the following page
    https://google.github.io/filament/webgl/tutorial_triangle.html
3. Check the console.

The following error is output to the console and the triangle is not displayed on the screen.
```
GL_INVALID_FRAMEBUFFER_OPERATION: Framebuffer is incomplete.
GL_INVALID_FRAMEBUFFER_OPERATION: Draw framebuffer is incomplete.
```

**Expected behavior**

The assumed result is a triangle on the screen and no error output to the console.

**Screenshots**

|Chrome 87.0|Firefox 82.0|
|:-----:|:-----:|
|![image](https://user-images.githubusercontent.com/1833979/99874487-b19cef00-2c2b-11eb-85af-c0b12c51d96d.png)|![image](https://user-images.githubusercontent.com/1833979/99874490-b792d000-2c2b-11eb-8a26-801bdab32fa9.png)|

**Desktop (please complete the following information):**
 - OS: Windows 10 Version 1909
 - GPU: Intel UHD Graphics 617
 - Backend: ANGLE
 - Browser: Chrome 87.0.4280.66

**Comment by romainguy** [comment]
I cannot reproduce. The samples work for me on Chrome 87 (macOS).

**Comment by prideout** [comment]
Most samples work, but the triangle demo does not, which is a dup of #3128.

**Comment by romainguy** [comment]
The triangle works for me :)

**Comment by cx20** [comment]
> I cannot reproduce. The samples work for me on Chrome 87 (macOS).

Yes. The problem seems to occur only in the Windows version of Chrome 87.
I also confirmed that the triangle appears in the case of Chrome 87 on macOS.

> Most samples work, but the triangle demo does not, which is a dup of #3128.

Hmm. I presented the triangle sample as a representative example, but neither the [other samples](https://google.github.io/filament/webgl/) nor [gltf-test](https://cx20.github.io/gltf-test/?engines=Filament) worked in my Windows environment with Chrome 87.

I have three Windows 10 environments and all have had similar events. The environments I've tried are as follows.
- MacBookAir + Windows 10 (BootCamp) + Chrome 87 + Intel UHD Graphics 617 
- ThinkPad X260 + Windows 10 + Chrome 87 + Intel HD Graphics 520
- Desktop PC + Windows 10 + Chrome 87 + GeForce GTX 660

**Comment by cx20** [comment]
This problem seems to be specific to the Windows version of `Chrome 87`.
Should I report a bug to [bugs.chromium.org](https://bugs.chromium.org/p/chromium/issues/list)?

**Comment by cx20** [comment]
I tried it out by embedding the debug output code into the following function in `filament.js`
- _glClearBufferfv()
- _glDrawBuffers()
- _glDrawElements()

The embedded code is as follows:
```javascript
console.log(_emscripten_glCheckFramebufferStatus(36160)); // 36160 is gl.FRAMEBUFFER
```

The results, traced in Spector.js, are as follows:

|WebGL2 API|Chrome 87 (Windows)|Firefox 83 (Windows)|
|:---------:|:---------:|:---------:|
|clearBufferfv|FRAMEBUFFER_INCOMPLETE_ATTACHMENT|FRAMEBUFFER_COMPLETE|
|drawBuffers|FRAMEBUFFER_INCOMPLETE_ATTACHMENT|FRAMEBUFFER_COMPLETE|
|drawElements|FRAMEBUFFER_INCOMPLETE_ATTACHMENT|FRAMEBUFFER_COMPLETE|

**Comment by cx20** [comment]
I've embedded the trace output code in the sample below to see where the problem is happening.
https://cx20.github.io/webgl-test/examples/filament/square/index.html

Upon investigation, I found that the problem occurred at the point of the call to `framebufferTexture2D()`.
Perhaps some parameter inconsistency was detected by the browser version upgrade, but the specific combination of parameters is not known.
The following case seems to cause problems when specifying `RGBA16F` in `internalFormat`, but as far as I can see from the [webglreport](https://webglreport.com/?v=2), the `EXT_color_buffer_half_float` extension seems to be supported.

|Chrome 87 (Windows)|Firefox 83 (Windows)|
|:---:|:----:|
|![image](https://user-images.githubusercontent.com/1833979/100935822-f716c180-3533-11eb-9083-04772b8ffb8c.png)|![image](https://user-images.githubusercontent.com/1833979/100935788-ed8d5980-3533-11eb-8f11-a5dd146c7320.png)|

**Comment by prideout** [comment]
Thanks @cx20, I have a tentative fix that I'm going to push momentarily.  Would be great if you could test it out for us.

**Comment by cx20** [comment]
@prideout Thanks! I have confirmed that the latest version of Filament can now view glTF files again.

[Filament v1.9.10rc + Duck.gltf result:](https://rawcdn.githack.com/cx20/gltf-test/22d0e9f82a88a08f549a105200afad190875e038/examples/filament/index.html?category=sampleModels&model=Duck&scale=1&type=glTF)
![image](https://user-images.githubusercontent.com/1833979/100941978-df443b00-353d-11eb-8549-e1f1126b8e7e.png)

**Comment by cx20** [comment]
I tried tracing again on `Chrome 87` and `Firefox 83` just to be sure.
In both cases, `RGBA16F` was set and it was confirmed that the error was resolved.

|internalFormat|Chrome 87 (Windows)|Firefox 83 (Windows)|
|:---:|:----:|:----:|
|RGBA16F|![image](https://user-images.githubusercontent.com/1833979/101000902-d6855080-35a1-11eb-9ff7-2420c359ba2a.png)|![image](https://user-images.githubusercontent.com/1833979/101000939-e0a74f00-35a1-11eb-87a1-3c28b1f8587b.png)|

**Comment by cx20** [comment]
@prideout Previously, the Filament `1.9.10rc`(Nightly Build) temporarily addressed this issue, but the release version is no longer working. Is there a version that works somewhere?

**Comment by prideout** [comment]
@cx20 It is known that MSAA does not work on WebGL with certain Chrome backends (#3128), we are working on a fix, this was never temporarily fixed (not intentionally anyway).  The triangle demo is the only demo that enables MSAA, so all our other web samples should work.  If the other samples do not work, please file a new bug.

**Comment by cx20** [comment]
@prideout Hmm. Currently, none of the Filament `1.9.10` samples in [gltf-test](https://cx20.github.io/gltf-test/) work. Is it a usage problem?

[Filament v1.9.10rc + Duck.gltf result:](https://rawcdn.githack.com/cx20/gltf-test/22d0e9f82a88a08f549a105200afad190875e038/examples/filament/index.html?category=sampleModels&model=Duck&scale=1&type=glTF)
![image](https://user-images.githubusercontent.com/1833979/100941978-df443b00-353d-11eb-8549-e1f1126b8e7e.png)

[Filament v1.9.10 + Duck.gltf result:](https://rawcdn.githack.com/cx20/gltf-test/72d25434bafbe3ce3f2eec3008a62a03c7bc112e/examples/filament/index.html?category=sampleModels&model=Duck&scale=1&type=glTF)
![image](https://user-images.githubusercontent.com/1833979/106785241-8a818580-6690-11eb-8303-a66a37ad096e.png)

The following sample seems to have the same error.
https://google.github.io/filament/webgl/tutorial_redball.html
https://google.github.io/filament/webgl/tutorial_suzanne.html
https://google.github.io/filament/webgl/parquet.html
https://google.github.io/filament/webgl/suzanne.html
https://google.github.io/filament/webgl/helmet.html

The environment is MacBookAir + Windows 10(BootCamp) + Chrome88.0.4324.146.

**Comment by prideout** [comment]
Ah, right, apologies.  Indeed, we did not merge the RGB16F fix  (#3334) into our 1.9.10 release candidate, as it was not considered critical fix for an internal upgrade at Google.  However the fix will definitely be included in our release next week.  Thank you for being patient with us.  :-)

**Comment by emackey** [comment]
Oh man.  VSCode has updated itself to use the version of Chrome that exhibits this behavior, causing the whole Filament preview window for glTF to go dark and report these errors.  I'll be looking forward to next week's release.

**Comment by amarav** [comment]
Can someone tell me how to compile filament sample to wasm in Windows? The guide describes only for macos and linux

**Comment by cx20** [comment]
@amarav If it is the version of Nightly Build, you can get wasm from the following.

https://github.com/google/filament/actions?query=workflow%3AWeb

![image](https://user-images.githubusercontent.com/1833979/107109140-f1fe2780-6880-11eb-8350-2aacce67ec17.png)

![image](https://user-images.githubusercontent.com/1833979/107109144-fa566280-6880-11eb-9256-3bda7c6dc527.png)

**Comment by cx20** [comment]
@emackey I have confirmed the same problem. The Filament sample seems to fail due to the VS Code chrome version being updated from 83 to 87.
I have confirmed that updating the version of Filament to `1.9.11rc` of Nightly Build in my local environment resolves the error.

![image](https://user-images.githubusercontent.com/1833979/107109211-89fc1100-6881-11eb-90ed-a9c9a0619ed1.png)

I too am looking forward to the next release.
